### PR TITLE
docs: Extend Hubble Performance Tuning Documentation

### PR DESCRIPTION
**Pull request by glrf**
Please ensure your pull request adheres to the following guidelines:

- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.

This change restructures the Hubble performance tuning guide and adds sections on the Hubble event queue and the monitor aggregation interval.

### CODE REVIEW

**Review by ysksuzuki — APPROVED** [review]

File: Documentation/operations/performance/tuning.rst [hunk 1]
```diff
@@ -339,10 +339,111 @@ performance. The overhead of Hubble is somewhere between 1-15% depending
...
+::
+
+   level=info msg="hubble events queue is processing messages again: NN messages were lost" subsys=hubble
+   level=warning msg="hubble events queue is full: dropping messages; consider increasing the queue size (hubble-event-queue-size) or provisioning more CPU" subsys=hubble
+
+By default the Hubble event queue size is ``#CPU * 1024``, or ``16384`` if your nodes have
+more than 16 CPU cores. If you encounter event bursts that result in dropped events,
+increasing this queue size might help. We recommend gradually doubling the queue length
+until the drops disappear. If you don't see any improvements after increasing the queue
+length to 128k, further increasing the event queue size unlikely to help.
```
Comment (qmonnet) [inline Documentation/operations/performance/tuning.rst]:
```suggestion
length to 128k, further increasing the event queue size is unlikely to help.
```

File: Documentation/operations/performance/tuning.rst [hunk 2]
```diff
@@ -339,10 +339,111 @@ performance. The overhead of Hubble is somewhere between 1-15% depending
...
+
+Be aware that increasing the Hubble event queue size will result in increased memory
+usage. Depending on your traffic pattern, increasing the queue size by ``10'000`` may
```
Comment (qmonnet) [inline Documentation/operations/performance/tuning.rst]:
```suggestion
usage. Depending on your traffic pattern, increasing the queue size by ``10,000`` may
```

File: Documentation/operations/performance/tuning.rst [hunk 3]
```diff
@@ -381,49 +482,53 @@ To enable eBPF Event Rate Limiting with a rate limit of 10,000 and a burst limit
              --set bpf.events.default.rateLimit=10000 \\
              --set bpf.events.default.burstLimit=50000
 
-If this is not sufficient, in order to optimize for maximum performance,
-you can disable Hubble:
+
+You can also choose to stop exposing event types in which you
```
Comment (qmonnet) [inline Documentation/operations/performance/tuning.rst]: No need for the additional blank line

```suggestion
You can also choose to stop exposing event types in which you
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
A couple of tiny nits/typos, but looks good otherwise, thanks!

**Review by qmonnet — APPROVED** [review]
Thanks!

File: Documentation/operations/performance/tuning.rst [hunk 4]
```diff
@@ -339,10 +339,111 @@ performance. The overhead of Hubble is somewhere between 1-15% depending
...
+           cilium install |CHART_VERSION| \\
+             --set bpf.events.monitorInterval="10s"
+
+    .. group-tab:: Helm
+
+       .. parsed-literal::
+
+           helm install cilium |CHART_RELEASE| \\
+             --namespace kube-system \\
+             --set bpf.events.monitorInterval="10s"
```
Comment (mkilchhofer) [inline Documentation/operations/performance/tuning.rst:440]: "bpf.**events**.monitorInterval"
and
"bpf.**events**.monitorInterval"
seems not existing. Is it a regression? I think the key `events` is obsolete and the interval can be controlled directly via `bpf.monitorInterval`: https://github.com/search?q=repo%3Acilium%2Fcilium%20%22bpf.monitorInterval%22&type=code

**Pull request merged**